pkg/oauth/handlers/default_auth_handler.go

```diff
@@ -116,6 +117,10 @@ func (authHandler *unionAuthenticationHandler) AuthenticationNeeded(apiClient au
 				w.WriteHeader(http.StatusUnauthorized)
 				ev := audit.AuditEventFrom(req.Context())
 				if ev != nil {
+					if ev.ResponseStatus == nil {
```

Comment:
When would that happen?

Illustrate on https://github.com/openshift/oauth-server/blob/5597860a7484ea3e680b2564a4ba89a83888ace3/vendor/k8s.io/apiserver/pkg/endpoints/filters/audit.go#L42

Reply:
Hypershift is running without an audit policy file. If I drop the audit policy file in the oauth-server on OCP, I am able to reproduce the issue.

`WithAudit` isn't being turned on, without an audit policy file, but there is a function that is being run always: https://github.com/openshift/oauth-server/blob/master/vendor/k8s.io/apiserver/pkg/endpoints/filters/audit_init.go#L41-L42